sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -51,6 +54,7 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] {
          """.stripMargin)
 
       val scanRelation = DataSourceV2ScanRelation(relation.table, scan, output)
+      scanRelation.setTagValue(PUSHED_FILTERS_TAG, pushedFilters)
```

Comment:
It will be convenient if `Scan` can report pushed filters itself. But I don't know how to design the API to make it work.

Here I just store the pushed filters in the  `DataSourceV2ScanRelation`, so that I can use it when creating v1 physical scan node later, which needs `pushedFilters` to do equality check.